main.go

```diff
@@ -222,13 +223,30 @@ func main() {
 	result, platformsDetected := scanner.GenerateScanResult(searchDir, hasSSHKey)
 
 	// Upload results
-	if resultClient != nil {
-		log.TInfof("Submitting results...")
-		if err := resultClient.uploadResults(result); err != nil {
-			failf("Could not send back results: %s", err)
+	var resultBytes []byte
+	shouldStoreResult := resultClient != nil || strings.HasPrefix(cfg.ResultSubmitURL, "path::")
+	if shouldStoreResult {
+		resultBytes, err = json.MarshalIndent(result, "", "\t")
```

Comment:
I know it is only whitespace but this will change the format of the result a bit. Just double checking that whatever component relies on this result in our system can still read it, right?

Reply:
This is the current format: https://github.com/bitrise-steplib/steps-project-scanner/blob/master/submitresult.go#L91
I didn't plan to change it with this PR.